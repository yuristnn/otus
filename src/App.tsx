import React, { useState } from 'react';
import Rows from './Rows';

const App: React.FC = () => {
    const [rows, setRows] = useState(3);
    const [cells, setCells] = useState(3);

    return (
        <div>
            <div>
                <p>Введите высоту поля:</p>
                <input
                    value={rows}
                    onChange={(e) => setRows(+e.target.value)}
                    type="number"
                />
            </div>
            <div style={{ marginBottom: 20 }}>
                <p>Введите ширину поля:</p>
                <input
                    value={cells}
                    onChange={(e) => setCells(+e.target.value)}
                    type="number"
                />
            </div>
            <table style={{ borderCollapse: 'collapse' }}>
                <tbody>
                    <Rows rows={rows} cells={cells} />
                </tbody>
            </table>
        </div>
    );
};

export default App;
