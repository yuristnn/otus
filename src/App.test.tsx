import React from 'react';
import App from './App';
import { shallow } from 'enzyme';

describe('App component test', () => {
    it('Should render App component', () => {
        const component = shallow(<App />);
        expect(component).toMatchSnapshot();
    });
});
