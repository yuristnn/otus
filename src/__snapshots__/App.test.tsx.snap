// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component test Should render App component 1`] = `ShallowWrapper {}`;
